### PR TITLE
Fixes #20: Pass access_token on after uploads.post()

### LINKED ISSUE #20: Upload Check unauthorized w/o strava_config
While the file upload works fine when passing the access_token explicitly instead of specifying it in the central strava_config file, the following status checks are unauthorized in this case.

The access token is not passed on via the arguments list from `strava.uploads.post()` to the subsequent `_check()` calls.

This does not prevent the upload of files, it only prevents the caller from determining whether and at which point the upload was successful. As a workaround, the access_token can be specified in the strava_config file, which is honored by the check calls.

### PR DESCRIPTION
**Pull request by TimoSta**
The access_token was not passed on to subsequent calls of `_check()` after
a file upload via `uploads.post()`. This caused authorization issues if
the token was not maintained in the central strava_config file but passed
in explicitly.

### CODE REVIEW

File: test/uploads.js [hunk 1]
```diff
@@ -6,37 +6,36 @@
...
-                    //(payload.city).should.be.exactly("Seattle");
-                }
-                else {
-                    console.log(err);
-                }
-
-                //done();
-            });
+        it('should upload a GPX file', function(done) {
+            new Promise((resolve, reject) => {
```
Comment (UnbounDev) [inline test/uploads.js]: This breaks the tests on node `v0.12.x` due to the use of arrow functions! We'd have to add babel support to the repo to support these changes in a non breaking way

```
Running "simplemocha:all" (simplemocha) task
Warning: /var/www/strava/test/uploads.js:12
            new Promise((resolve, reject) => {
                                          ^^
Unexpected token => Used --force, continuing.
```

**Comment by TimoSta** [comment]
I think the better choice would be to not use ES6 features that require babel or polyfills to run with 0.12.x to keep the library as lean as possible.
If you agree, I'm more than happy to replace the arrow functions.

As far as I know, 0.12 is in maintenance until sometime late 2016. After that, it may be a good idea to drop support for this node version with a new major version of the library and refactor the library to use more ES6 syntax. This would definitely make the code more readable and concise.

**Comment by UnbounDev** [comment]
I'm all for that - when the respective branches have been edited to pre ES6 compliance I'll continue w/ the merge+publish

**Comment by TimoSta** [comment]
Replaced the arrow functions with regular functions. The unit tests are now running fine all the way down to v0.10.x.

Also added a Promise polyfill to allow using promises in the unit tests without breaking compatibility with older node versions.

**Pull request merged**